analysers/analyser_osmosis_polygon_intersects.py

```diff
@@ -0,0 +1,316 @@
+  END AS landusevalue,
+  tags,
+  FALSE AS is_multipolygon
+FROM
+  ways
+WHERE
+  is_polygon AND
+  tags != ''::hstore AND
+  (tags?'natural' OR tags?'landuse') AND
+  -- Leave minipolygons for analyser osmosis_polygon_small
+  ST_Area(ST_MakePolygon(ST_Transform(linestring, {proj}))) >= 20
+UNION ALL
+SELECT
+  ways.id,
+  ways.linestring,
+  -- Better would be to parse inner and outer segments and create an SQL multipolygon
+  -- For now, just use a 0.0000000001m buffer zone around the linestring
+  ST_Buffer(ways.linestring, 0.0000000001, 'quad_segs=1 endcap=flat join=bevel') AS poly,
+  CASE
+    WHEN r.tags?'landuse' THEN 'landuse'
+    WHEN r.tags?'natural' THEN 'natural'
+  END AS landusekey,
+  CASE
+    WHEN r.tags?'landuse' THEN r.tags->'landuse'
+    WHEN r.tags?'natural' THEN r.tags->'natural'
+  END AS landusevalue,
+  r.tags,
+  TRUE AS is_multipolygon
+FROM
+  relation_members
+  LEFT JOIN relations AS r ON
+    relation_id = r.id
+  JOIN ways ON
+    member_role = 'outer' AND
+    member_id = ways.id AND
+    member_type = 'W'
+WHERE
+  r.tags->'type' = 'multipolygon' AND
+  (r.tags?'natural' OR r.tags?'landuse')
+"""
+
+sql12 = """
+CREATE INDEX idx_mobilityway_linestring ON mobility_ways USING GIST(linestring);
+CREATE INDEX idx_landusage_linestring ON landusage USING GIST(linestring);
+CREATE INDEX idx_landusage_poly ON landusage USING GIST(poly);
+"""
+
+sql13 = """
+CREATE TEMP TABLE intersections AS
+SELECT
+  mobility_ways.id AS mobility_way_id,
+  mobility_ways.linestring AS mobility_way_linestring,
+  mobility_ways.type AS mobility_way_type,
+  landusage.id AS landusage_way_id,
+  landusage.poly AS landusage_poly,
+  landusage.landusekey,
+  landusage.landusevalue,
+  CONCAT(mobility_ways.type, '=', mobility_ways.tags->mobility_ways.type) AS tag_way,
+  CONCAT(landusage.landusekey, '=', landusage.landusevalue) AS tag_polygon
+FROM
+  mobility_ways
+  JOIN landusage ON
+    (
+      NOT landusage.tags?'layer' AND NOT mobility_ways.tags?'layer' OR
+      (
+        landusage.tags?'layer' AND mobility_ways.tags?'layer' AND
+        landusage.tags->'layer' = mobility_ways.tags->'layer'
+      )
+    ) AND
+    landusage.linestring && mobility_ways.linestring AND
+    -- Only find ways where the highway/railway/waterway/aeroway actually enters the
+    -- polygon. Exclude cases where the polygon is tied to the way without going inside.
+    ST_Intersects(landusage.poly, mobility_ways.linestring) AND
+    NOT ST_Touches(landusage.poly, mobility_ways.linestring) AND
+    (
+      -- Ignore very minor overlaps. The distance between the center of the overlap and one of the ways must be at least 1m
+      -- Can't do this for multipolygons (without parsing all inner/outer segments) hence disabling this simplification for those
+      landusage.is_multipolygon OR
+      ST_Distance(ST_Centroid(ST_Transform(ST_Intersection(landusage.poly, mobility_ways.linestring), {proj})), ST_Transform(mobility_ways.linestring, {proj})) > 1 OR
```

Comment:
The intersection could be a multi, and the centroid not a reliable value.

Reply:
You're right!
I couldn't think of a better, simple solution to detect how far a linestring penetrates into an area*. Hence, I've removed this "ignore small overlaps" code. They should be fixed anyway; they're just a bit more... spammy ;) (I only had this block to reduce the number of issues to empathize the most important ones with big overlaps.)

* The intersection result of linestring + polygon is also a linestring, so anything area-based will fail